doc/sfp-cmis/CMIS-custom-SI-settings.md

```diff
@@ -0,0 +1,267 @@
+
+### References
+
+#### Table 2: References
+
+| **Document**                                            | **Location**  |
+|---------------------------------------------------------|---------------|
+| CMIS v5 | [CMIS5p0.pdf](http://www.qsfp-dd.com/wp-content/uploads/2021/05/CMIS5p0.pdf) |
+
+### About This Manual
+This is a high-level design document describing the way to apply custom SI settings for CMIS supported modules.
+
+## 1 Introduction and Scope
+Certain high-speed QSFP_DD, OSFP and QSFP modules require Signal Integrity (SI) settings to match platform media settings in order to achieve link stability, right tunning and optimal performance.
+
+### 1.1 Clause from CMIS5p0 spec for Signal Integrity
+Excerpt from CMIS5.0 spec providing definition of Signal Integrity:
+
+![image](https://user-images.githubusercontent.com/115578705/236561523-8999b615-b271-4e28-9fbe-d0c9d414bdb8.png)
+
+These SI settings can vary based on combination of the module vendor plus platform vendor. The module will have default TX/RX SI settings programmed in its EEPROM by module vendor, but platform vendor has provision to overwrite these settings to match their platform requirements. 
+The host can apply new TX SI settings when TX Input Adaptive EQ is disabled for all TX Data Path lanes but RX SI settings can be applied directly. These TX/RX setting should be applied with Explicit Control bit is set to 1.
+
+### 1.2 Clause from CMIS5p0 spec for Explicit Control
+Excerpt from CMIS5.0 spec providing definition of Explicit Control:
+
+![image](https://user-images.githubusercontent.com/115578705/236561421-d960d243-cd26-4087-88fe-c621867ffaa7.png)
+
+## 2 Requirements
+This feature would be enabled per platform basis. If platform wants to use this feature, they would need to provide optics_si_setting.json file during init for XCVRD to parse it. The SI parameters can be vendor and module specific. The vendor can populate desired SI param values in a JSON file. Provide an approach in the CMIS state machine to generate and apply host defined SI parameters to module eeprom. The Modules that do not support CMIS and not part of CMIS state machine are not in the scope of this document. 
+
+![image](https://user-images.githubusercontent.com/115578705/236575703-aea7f377-ba5e-4e96-b18e-920f93e19774.png)
+
+## 3 Architecture Design
+The SI media setting file optics_si_setting.json needs to be defined by each platform_vendor that will need SI settings. All SKUs of the platform will share the same optics_si_setting.json file. If no file is found, then this mechanism will be ignored.
+
+This file will have TX, RX setting blocks, and each block will have two subblocks: the first is global level setting and the next is port level setting. These subblocks will eventually contain per-lane SI parameter setting values based on the type of vendor and speed that are expected to be programmed. The SI settings will not depend on cable length.
```

Comment:
@AnoopKamath why do you say SI setting will not depend upon cable length? We have NPU side SI setting different for different cable lengths.

Reply:
SI settings will only deal with the electrical side of the cable, so that is not affected by the length on the optical side but even if the values are different, it will have different part number and section in optics_si_settings.json.